py_rql/filter.py

```diff
@@ -0,0 +1,93 @@
+#
+#  Copyright © 2022 Ingram Micro Inc. All rights reserved.
+#
+from cachetools import LFUCache
+
+from py_rql import parse
+from py_rql.cast import get_default_cast_func_for_type
+from py_rql.constants import FilterLookups, FilterTypes, RESERVED_FILTER_NAMES
+from py_rql.exceptions import (
+    RQLFilterLookupError,
+    RQLFilterNotFoundError,
+)
+from py_rql.transformer import RQLToFunctionTransformer
+
+
+class Filter:
+
+    FILTERS = []
+
+    def __init__(self):
+        self._cache = LFUCache(maxsize=1000)
+        self._filters = self.init_filters()
+
+    def init_filters(self):
+        filters = {}
+        for filter_decl in self.FILTERS:
```

Comment:
Please, insert blank lines between logically separated pieces of codes everywhere. Currently, it's extremely hard for me to read. Just for example:
```python
            filter_name = filter_decl['filter']
            e = f"'{filter_name}' is a reserved filter name."
            assert filter_name not in RESERVED_FILTER_NAMES, e

            filter_type = filter_decl.get('type', FilterTypes.STRING)
            default_lookups = self.default_filter_lookups_for_type(filter_type)
            lookups = filter_decl.get('lookups', default_lookups)
            e = f"Invalid lookups for filter '{filter_name}' of type '{filter_type}': '{lookups}'"
            assert set(lookups).issubset(default_lookups), e

            cast_func = filter_decl.get(
                'cast_func',
                get_default_cast_func_for_type(filter_type),
            )
            e = f"Invalid cast function for filter '{filter_name}'."
            assert callable(cast_func), e

            filters[filter_name] = {
                'type': filter_type,
                'lookups': lookups,
                'cast_func': cast_func,
            }
....
```